### PR TITLE
Requests Fix

### PR DESCRIPTION
**Pull request by mayankamencherla**
Requiring Requests class only if not already defined

### CODE REVIEW

File: Razorpay.php [hunk 1]
```diff
@@ -1,10 +1,13 @@
 <?php
 
-// Include Requests
-require_once __DIR__.'/libs/Requests-1.6.1/library/Requests.php';
+// Include Requests only if not already defined
```
Comment (captn3m0) [inline Razorpay.php]: The only concern I have with this is that Requests could have been defined as any class, not specifically the class we are looking for. So far we've dealt with conflicts in only one place: WordPress, where commenting out this line works as a workaround, but if someone happens to have a Requests class defined that is not what we are expecting, this will throw up weirder errors than class defined again.

Might work (since we haven't seen a conflict with another class called Requests yet) out well though.

Comment (mayankamencherla) [inline Razorpay.php]: Cool. Let us keep it as is for now, and lets see how things work.

File: composer.json [hunk 2]
```diff
@@ -28,12 +28,13 @@
     "homepage": "https://docs.razorpay.com",
     "license": "MIT",
     "require": {
-        "php": ">=5.3.2",
+        "php": ">=5.3.3",
```
Comment (shashankkumar) [inline composer.json]: Why this change?

Comment (mayankamencherla) [inline composer.json]: php 5.3.3 is a requirement for phpunit 4.8.0 
https://github.com/sebastianbergmann/phpunit/wiki/Release-Announcement-for-PHPUnit-4.8.0#requirements

File: Razorpay.php [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+if (class_exists('Requests') === false)
+{
+    require_once __DIR__.'/libs/Requests-1.6.1/library/Requests.php';
 
-// Register requests autoloader
-Requests::register_autoloader();
+    // Register requests autoloader
+    Requests::register_autoloader();
+}
```
Comment (shashankkumar) [inline Razorpay.php]: We should throw an exception in the else clause giving clear indication of why it's failed otherwise, this may fail later in the code and it will be difficult to debug.

File: composer.json (hunk 2 above)
Comment (shashankkumar) [inline composer.json]: Okay with this change.

File: Razorpay.php (hunk 3 above)
Comment (mayankamencherla) [inline Razorpay.php]: @shashankkumar But, if `Requests` is already defined, (like in the case of wordpress), we shouldn't throw an exception, right? Instead, the purpose of this check is to ensure we are not loading a class that is already existing within the framework, and in that case, throwing an exception would not be the correct behaviour, right?

Comment (captn3m0) [inline Razorpay.php]: I like the idea of throwing an exception. Require requests.php if the class is not defined. If it is defined, assume it to be the exact Requests library we want. Call `register_autoloader` assuming that, and if it fails, throw an exception saying "Requests library, found but didn't match". We could also do a check against the Requests lib version. (>=1.6 should be good for us).

Comment (shashankkumar) [inline Razorpay.php]: agree with @captn3m0. Please implement this and circle back.

File: Razorpay.php [hunk 4]
```diff
@@ -1,10 +1,29 @@
...
+    }
+}
+catch (\Exception $e)
+{
+    throwRequestsException();
+}
 
-// Register requests autoloader
-Requests::register_autoloader();
+function throwRequestsException()
```
Comment (captn3m0) [inline Razorpay.php]: let us not touch the global namespace

Comment (mayankamencherla) [inline Razorpay.php]: Fixed
